fetch/mesh_data.go

```diff
@@ -270,9 +269,7 @@ func (f *Fetch) peerEpochInfoStreamed(ctx context.Context, peer p2p.Peer, epochB
 	if err := f.meteredStreamRequest(
 		ctx, atxProtocol, peer, epochBytes,
 		func(ctx context.Context, s io.ReadWriter) (int, error) {
-			return server.ReadResponse(s, func(respLen uint32) (n int, err error) {
-				return codec.DecodeFrom(s, &ed)
-			})
+			return readIDSlice(s, &ed.AtxIDs, maxEpochDataAtxIDs)
```

Comment:
Would this also work?

```go
return server.ReadResponse(rw, func(respLen uint32) (n int, err error) {
	if respLen > maxEpochDataAtxIDs*types.ATXIDSize {
		return 0, errors.New("epochInfoStream: response too big")
	}
	return codec.DecodeFrom(io.LimitReader(rw, int64(respLen)), &ed)
})
```

Reply:
As discussed: `readIDSlice` also verifies consistency between outer `Response` length and the slice length